modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileDumper.java

```diff
@@ -326,13 +344,27 @@ private void writeToShapefile(
     }
 
     /**
-     * Allows subsclasses to perform extra actions against a shapefile that was completely written
+     * Allows subsclasses to perform extra actions against a shapefile that was completely written.
+     *
+     * <p>By default, this method calls {@link #shapefileDumped(SimpleFeatureType)}.
      *
      * @param fileName
      * @param remappedSchema
      */
     protected void shapefileDumped(String fileName, SimpleFeatureType remappedSchema)
             throws IOException {
+        shapefileDumped(remappedSchema);
+    }
+
+    /**
+     * Allows subsclasses to perform extra actions against a shapefile that was completely written.
+     *
+     * <p>This method will not be called if {@link #shapefileDumped(String, SimpleFeatureType)} is
+     * overwritten.
+     *
+     * @param remappedSchema
+     */
+    protected void shapefileDumped(SimpleFeatureType remappedSchema) throws IOException {
```

Comment:
The PR certainly got more convoluted but does not break backwards compatibility.
Just a question about this method, why is it here? The original method is the only one still being called, and this one does not add anything, actually, it removed one parameter that subclasses could just ignore when using the pre-existing method? Makes sense? What did I miss?

Besides this, the rest looks good. Did you try out a GeoServer build?

Reply:
I'm confused myself, I thought this was the original method and I was preserving it for backwards compatibility. I will remove it again. Yes, this time geoserver builds

Reply:
Then after removing the method go ahead and merge